tests/utils.py

```diff
@@ -58,62 +58,68 @@ def run_with_reference(
         ):
 
 
-    res, config = execute_pysvzerod(test_config, "solver")
+    res = pysvzerod.simulate(test_config)
+
+    if res.shape[1] >= 6:
+        # we have a result with fields [name, time, p_in, p_out, q_in, q_out] SOME HAVE GREATTER LENGTH NEED TO ADDRESS
+        pressure_cols = ["pressure_in", "pressure_out"]
+        flow_cols = ["flow_in", "flow_out"]
+
+        bool_df = pd.DataFrame(index=res.index, columns=res.columns, dtype=bool)
+
+        for col in res.columns:
+            tol = RTOL_PRES if col in pressure_cols else RTOL_FLOW if col in flow_cols else None
+            if tol is not None:
+                rel_diff = np.abs(res[col] - ref[col]) - tol - tol * np.abs(ref[col])
```

Comment:
Since this is key to the check, put this comparison in a "diff" function and then call it here and below